core/transfer/image/imagestore.go

```diff
@@ -212,9 +215,56 @@ func (is *Store) ImageFilter(h images.HandlerFunc, cs content.Store) images.Hand
 	return h
 }
 
-func (is *Store) Store(ctx context.Context, desc ocispec.Descriptor, store images.Store) ([]images.Image, error) {
+// checkUnpackAndAddPlatformLabel checks if the requested image has been successfully
+// unpacked for the requested platforms. If yes, platform image labels are added
+// to indicate what platforms the image has been pulled for. This becomes important
+// to support image pull per runtime class where an image could be pulled for different
+// platforms.
+func (is *Store) checkUnpackAndAddPlatformLabel(ctx context.Context, desc ocispec.Descriptor, store images.Store, content content.Provider) error {
+	for _, platformSpec := range is.platforms {
+		platformMatcher := platforms.Only(platformSpec)
+
+		// Check if required layers were pulled and unpacked successfully.
+		manifest, err := images.Manifest(ctx, content, desc, platformMatcher)
+		if err != nil {
+			log.G(ctx).Infof("Image %v not unpacked for platform %v. Skip adding image label.", is.imageName, platformSpec)
+		}
+		imageLayers := []ocispec.Descriptor{}
+		for _, ociLayer := range manifest.Layers {
+			if images.IsLayerType(ociLayer.MediaType) {
+				imageLayers = append(imageLayers, ociLayer)
+			}
+		}
+
+		diffIDs, err := images.RootFS(ctx, content, manifest.Config)
+		if err != nil {
+			return fmt.Errorf("error while getting diffTDs for %v", is.imageName)
+		}
+		if len(diffIDs) != len(imageLayers) {
+			return errors.New("mismatched image rootfs and manifest layers")
+		}
+
+		// Add platform image label
+		platform := platforms.Format(platformSpec)
+		platformImageLabel := fmt.Sprintf(ctrdlabels.PlatformLabelFormat, ctrdlabels.PlatformLabelPrefix, platform)
```

Comment:
I'm not really sure if this label here makes sense, @kiashok.

For linux x86_64 we're labeling the image with:
```
containerd.io/imagePulledForPlatform.linux/amd64=linux/amd64
```

And the `.linux/amd64=linux/amd64`  part seems redundant.

Labeling it with the snapshotter / runtime-handler wouldn't be better? Something like:
```
containerd.io/imagePulledForPlatform.linux/amd64=nydus
```
or 
```
containerd.io/imagePulledForPlatform.linux/amd64=kata-qemu
```

Reply:
Please refer to responses https://github.com/containerd/containerd/pull/9832#issuecomment-2070827361 , https://github.com/containerd/containerd/pull/9832#discussion_r1575277817 , https://github.com/containerd/containerd/pull/9832#discussion_r1575286734. 
Also there is a format for the platform label as defined in github.com/containerd/platforms repo and that is what will be used.